source/geod24/LocalRest.d

```diff
@@ -1,30 +1,68 @@
 /*******************************************************************************
 
-    Provides utilities to mock an async REST node in unittests
-
-    Using `vibe.web.rest` allows to cleanly separate business code
-    from network code, as implementing an interface is all that's
-    needed to create a server.
-
-    However, in order for tests to simulate an asynchronous system
-    accurately, multiple nodes need to be able to run concurrently.
+    Provides utilities to mock a network in unittests
+
+    This module is based on the idea that D `interface`s can be used
+    to represent a server's API, and that D `class` inheriting this `interface`
+    are used to define the server's business code,
+    abstracting away the communication layer.
+
+    For example, a server that exposes an API to concatenate two strings would
+    define the following code:
+    ---
+    interface API { public string concat (string a, string b); }
+    class Server : API
+    {
+        public override string concat (string a, string b)
+        {
+            return a ~ b;
+        }
+    }
+    ---
+
+    Then one can use "generators" to define how multiple process communicate
+    together. One such generator, that pioneered this design is `vibe.web.rest`,
+    which allows to expose such servers as REST APIs.
+
+    `localrest` is another generator, which uses message passing and threads
+    to create a local "network".
+    The motivation was to create a testing library that could be used to
+    model a network at a much cheaper cost than spawning processes
+    (and containers) would be, when doing integration tests.
+
+    Control_Interface:
+    When instantiating a `RemoteAPI`, one has the ability to call foreign
+    implementations through auto-generated `override`s of the `interface`.
+    In addition to that, as this library is intended for testing,
+    a few extra functionalities are offered under a control interface,
+    accessible under the `ctrl` namespace in the instance.
+    The control interface allows to make the node unresponsive to one or all
+    methods, for some defined time or until unblocked.
+    See `sleep`, `filter`, and `clearFilter` for more details.
+
+    Event_Loop:
+    Server process usually needs to perform some action in an asynchronous way.
+    Additionally, some actions needs to be completed at a semi-regular interval,
+    for example based on a timer.
+    For those use cases, a node should call `runTask` or `sleep`, respectively.
```

Comment:
I would also make a note not to call vibe.d's runTask / wait by accident, or just to be sure not to mix localrest & vibe.d code.

Reply:
Added clarification